backend/src/Database.hs

```diff
@@ -171,9 +170,17 @@ toFilterExpression report spec = foldr ((&&.) . fromMaybe (val True)) (val True)
     loserFilter = toListFilter report GameReportLoserId <$> spec.losers
     turnsFilter = toInequalityFilter report GameReportTurns <$> spec.turns
     victoryFilter =
-      spec.victory
-        <&> foldr1 (||.)
-        . fmap (\(s, v) -> (report ^. GameReportSide) ==. val s &&. (report ^. GameReportVictory) ==. val v)
+      spec.victory <&> \(v :| vs) ->
+        foldr
+          ( (||.)
+              . ( \case
+                    VictoryKindFilter k -> report ^. GameReportVictory ==. val k
+                    VictorySideFilter s -> report ^. GameReportSide ==. val s
+                    VictoryComboFilter s k -> (report ^. GameReportSide) ==. val s &&. (report ^. GameReportVictory) ==. val k
+                )
+          )
+          (val False)
+          (v : vs)
```

Comment:
This works technically but looking at the diff I think I can make it cleaner, will work on it